docs: Clean up incomplete recipes and clarify Kubernetes-only focus

BenHamm opened the pull request:
## 🎯 Objective

Clean up the `recipes/` directory to focus exclusively on production-ready Kubernetes deployments. Remove incomplete configurations and clarify that benchmark recipes are tools for users, not published performance results.

**Related PR to release/0.6.1:** #4145

https://github.com/ai-dynamo/dynamo/pull/3772/files

---

## 📊 Summary of Changes

**34 files changed: +210 insertions, -1,587 deletions**

### Deleted Content
- **5 incomplete model directories** (no K8s `deploy.yaml` manifests):
  - `deepseek-r1-distill-llama-8b/`
  - `gemma3/`
  - `llama4/`
  - `qwen2-vl-7b-instruct/`
  - `qwen3/` (kept `qwen3-32b-fp8/` which is complete)

- **`run.sh` script** (228 lines) - non-K8s automation tool

- **12 standalone engine config YAMLs** from `deepseek-r1/trtllm/`:
  - `agg/simple/`, `agg/mtp/`, `agg/wide_ep/` (all subdirs)
  - `disagg/simple/`, `disagg/mtp/`
  - `disagg/wide_ep/*.yaml` (standalone configs only)
  - Kept: `disagg/wide_ep/gb200/` (has complete K8s manifests)

### Added/Modified Content
- **Comprehensive new `recipes/README.md`** with:
  - Complete recipe table showing **Deployment** and **Benchmark Recipe** availability
  - Prerequisites section linking to correct K8s installation guides
  - Step-by-step deployment instructions
  - Troubleshooting section

- **`gpt-oss-120b/trtllm/disagg/README.md`** - Documents incomplete recipe status

---

## 📦 What Remains

**4 Models | 10 Complete Deployments | 7 with Benchmark Recipes**

| Model | Framework | Recipes | Status |
|-------|-----------|---------|--------|
| **llama-3-70b** | vLLM | agg, disagg-single-node, disagg-multi-node | ✅ All with deploy.yaml + perf.yaml |
| **qwen3-32b-fp8** | TensorRT-LLM | agg, disagg | ✅ All with deploy.yaml + perf.yaml |
| **gpt-oss-120b** | TensorRT-LLM | agg, disagg* | ✅ agg complete; disagg documented as incomplete |
| **deepseek-r1** | SGLang + TensorRT-LLM | sglang/disagg-8gpu, sglang/disagg-16gpu, trtllm/disagg/wide_ep/gb200 | ✅ 2 functional, 1 complete |

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **Documentation**
  * Comprehensively rewrote core recipe documentation with structured guidance on prerequisites, deployment steps, customization, and troubleshooting
  * Added documentation for GPT-OSS-120B disaggregated deployment mode

* **Chores**
  * Removed multiple configuration files across model recipes (deepseek, gemma3, llama4, qwen variants)
  * Removed automated deployment orchestration script

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

BenHamm committed:
recipes: Clean up incomplete recipes and clarify Kubernetes-only focus

Remove incomplete model directories and non-Kubernetes configurations to
streamline the recipes directory for production Kubernetes deployments.

Changes:
- Remove 5 incomplete model directories (deepseek-r1-distill-llama-8b,
  gemma3, llama4, qwen2-vl-7b-instruct, qwen3) that lack proper
  Kubernetes deployment manifests
- Delete run.sh script (non-Kubernetes automation tool)
- Remove standalone engine config YAMLs from deepseek-r1/trtllm that
  were not wrapped in Kubernetes manifests
- Document incomplete gpt-oss-120b disagg recipe with README explaining
  missing components

README improvements:
- Restructure Available Recipes table with 'Deployment' and 'Benchmark
  Recipe' columns to clarify that perf.yaml files are tools for users
  to run benchmarks, not published performance results
- Add comprehensive quick start guide with prerequisites
- Link to correct Kubernetes deployment guides
- Add troubleshooting section
- Remove extraneous links (docs.nvidia.com, license section)

Result: 4 models with 10 complete deployment recipes (7 with benchmark
scripts), focused exclusively on Kubernetes deployments.

Signed-off-by: Ben Hamm <ben.hamm@gmail.com>

BenHamm committed:
Remove emojis from README section headings

Address feedback to make the README less AI-generated looking by removing
decorative emojis from section headings while keeping status indicators
(✅ ❌) in tables and content.

Signed-off-by: Ben Hamm <ben.hamm@gmail.com>

athreesh commented on recipes/README.md:
can you hyperlink files in the Available Recipes table?

athreesh reviewed (approved):
Added a comment about hyperlinks but otherwise LGTM

BenHamm committed:
docs: add hyperlinks to recipe table for easier navigation

BenHamm committed:
docs: fix trailing whitespace in README

BenHamm commented:
✅ Addressed feedback:
- Added hyperlinks to the Recipes table (all model names now link to their recipe directories)
- Fixed trailing whitespace issue
- Ready for review

biswapanda committed:
restore trtllm configs in examples/trtllm

nv-anants reviewed (changes requested):
can you update all the paths in the repo for moved files as well please. example, these all will need update for llama files - 
```
git grep recipes/llama4/trtllm
docs/backends/trtllm/llama4_plus_eagle.md:31:* Make sure the (`eagle3_one_model: true`) is set in the LLM API config inside the `recipes/llama4/trtllm/eagle` folder.
docs/backends/trtllm/llama4_plus_eagle.md:55:export ENGINE_CONFIG="/mnt/recipes/llama4/trtllm/eagle/eagle_agg.yml"
docs/backends/trtllm/llama4_plus_eagle.md:63:export PREFILL_ENGINE_CONFIG="/mnt/recipes/llama4/trtllm/eagle/eagle_prefill.yaml"
docs/backends/trtllm/llama4_plus_eagle.md:65:export DECODE_ENGINE_CONFIG="/mnt/recipes/llama4/trtllm/eagle/eagle_decode.yaml"
docs/backends/trtllm/multimodal_support.md:30:export AGG_ENGINE_ARGS=./recipes/llama4/trtllm/multimodal/agg.yaml
docs/backends/trtllm/multinode/multinode-multimodal-example.md:37:> sed -i 's/backend: DEFAULT/backend: default/g' /mnt/recipes/llama4/trtllm/multimodal/prefill.yaml /mnt/recipes/llama4/trtllm/multimodal/decode.yaml
docs/backends/trtllm/multinode/multinode-multimodal-example.md:103:# export PREFILL_ENGINE_CONFIG="/mnt/recipes/llama4/trtllm/multimodal/prefill.yaml"
docs/backends/trtllm/multinode/multinode-multimodal-example.md:104:# export DECODE_ENGINE_CONFIG="/mnt/recipes/llama4/trtllm/multimodal/decode.yaml"
```

tanmayv25 committed:
Fix the TRTLLM engine config locations and references

tanmayv25 committed:
Fix the multinode documentation

tanmayv25 committed:
Fix the reference directory

nv-anants commented on examples/basics/multinode/trtllm/README.md:
```diff
@@ -0,0 +1,20 @@
+
+# Example: Multi-node TRTLLM Workers with Dynamo on Slurm
+
+See [here](docs/backends/trtllm/multinode) for how to setup this example.
```
```suggestion
See [here](/docs/backends/trtllm/multinode) for how to setup this example.
```

nv-anants commented on recipes/README.md:
````diff
@@ -1,297 +1,271 @@
-pkill -f "kubectl port-forward"
+**Container Images:**
+```yaml
+image: nvcr.io/nvidia/ai-dynamo/vllm-runtime:0.6.1
````
lets not commit any specific release tags to readme please. ask users to pick latest release

tanmayv25 commented on docs/backends/trtllm/multinode/multinode-examples.md:
````diff
@@ -136,7 +138,7 @@ follow these steps below to launch an **aggregated** deployment across 4 nodes:
 ```bash
 # Default set in srun_aggregated.sh, but can customize here.
-# export ENGINE_CONFIG="/mnt/recipes/deepseek-r1/trtllm/agg/wide_ep/wide_ep_agg.yaml"
+# export ENGINE_CONFIG="/mnt/examples/backends/trtllm/engine_configs/deepseek-r1/agg/wide_ep/wide_ep_agg.yaml"
````
Yes. This was a miss before. The instructions here should have been modified when this example doc was moved from `components/backends/trtllm/multinode` to `docs/backend/trtllm/multinode`.

tanmayv25 committed:
Apply suggestions from code review

Co-authored-by: Anant Sharma <anants@nvidia.com>
Signed-off-by: Tanmay Verma <tanmay2592@gmail.com>

tanmayv25 committed:
Add placeholder for the version

atchernych committed:
Update README.md with GAIE

Signed-off-by: atchernych <atchernych@nvidia.com>

tanmayv25 committed:
Add a new column
